src/monitors/turbFlowStats.cpp

```diff
@@ -0,0 +1,268 @@
+#include "monitors/turbFlowStats.hpp"
+#include "io/interval/fixedInterval.hpp"
+#include "petscmath.h"
+#include "solver/range.hpp"
+
+using tp = ablate::eos::ThermodynamicProperty;
+using fLoc = ablate::domain::FieldLocation;
+typedef ablate::solver::Range Range;
+
+ablate::monitors::TurbFlowStats::TurbFlowStats(const std::vector<std::string> nameIn, const std::shared_ptr<ablate::eos::EOS> eosIn, std::shared_ptr<io::interval::Interval> intervalIn)
+    : fieldNames(nameIn), eos(eosIn), interval(intervalIn ? intervalIn : std::make_shared<io::interval::FixedInterval>()) {}
+
+PetscErrorCode ablate::monitors::TurbFlowStats::MonitorTurbFlowStats(TS ts, PetscInt step, PetscReal crtime, Vec u, void* ctx) {
+    PetscFunctionBeginUser;
+    PetscErrorCode ierr;
+
+    // Loads in context
+    auto monitor = (ablate::monitors::TurbFlowStats*)ctx;
+
+    if (monitor->interval->Check(PetscObjectComm((PetscObject)ts), step, crtime)) {
+        std::vector<Vec> vec(monitor->fieldNames.size(), nullptr);
+        std::vector<IS> vecIS(monitor->fieldNames.size(), nullptr);
+        std::vector<DM> subDM(monitor->fieldNames.size(), nullptr);
+        for (int f = 0; f < (int)monitor->fieldNames.size(); f++) {
+            const auto& field = monitor->GetSolver()->GetSubDomain().GetField(monitor->fieldNames[f]);
+
+            ierr = monitor->GetSolver()->GetSubDomain().GetFieldGlobalVector(field, &vecIS[f], &vec[f], &subDM[f]);
+            CHKERRQ(ierr);
+        }
+
+        DM solDM;
+        Vec solVec;
+        solVec = monitor->GetSolver()->GetSubDomain().GetSubSolutionVector();
+        solDM = monitor->GetSolver()->GetSubDomain().GetDM();
+
+        //! Get relevant data
+        // Get cell range (just keep data in this struct and read out as necessary).
+        Range cellRange;
+        monitor->GetSolver()->GetCellRange(cellRange);
+
+        // Extract the solution global array
+        const PetscScalar* solDat;
+        ierr = VecGetArrayRead(solVec, &solDat);
+        CHKERRQ(ierr);
+
+        // Get the array of turbulent flow data
+        PetscScalar* turbDat;
+        ierr = VecGetArray(monitor->turbVec, &turbDat);
+        CHKERRQ(ierr);
+
+        // Get the timestep from the TS
+        PetscReal dt;
+        ierr = TSGetTimeStep(ts, &dt);
+        CHKERRQ(ierr);
+
+        //! Iterator guide
+        // f - field iterator
+        // c - cell iterator
+        // p - field component iterator
+        const PetscScalar* fieldDat;
+        for (int f = 0; f < (int)monitor->fieldNames.size(); f++) {
+            // Extract the field vector global array
+            ierr = VecGetArrayRead(vec[f], &fieldDat);
+            CHKERRQ(ierr);
+
+            //! Compute measures
+            for (int c = cellRange.start; c < cellRange.end; c++) {
```

Comment:
Would you mind changing this to PetscInt?  That would handle cases with very large number of cells.